githubapp/client_creator.go

```diff
@@ -132,6 +134,13 @@ func WithClientUserAgent(agent string) ClientOption {
 	}
 }
 
+// WithClientCache sets the http cache per constructed client
+func WithClientCache(cacheFunc interface{}) ClientOption {
+	return func(c *clientCreator) {
+		c.cacheFunc = cacheFunc.(func() *httpcache.Cache)
```

Comment:
This part isn't yet working:
```
interface conversion: interface {} is *httpcache.MemoryCache, not func() *httpcache.Cache
```

Reply:
I'd recommend specifying this signature as a `func() httpcache.Cache` and then wrapping the constructor in an anonymous function:

```go
WithClientCache(func() httpcache.Cache { return httpcache.NewMemoryCache() })
```

It's extra code, but I think it's the only way to satisfy the type system and still accept arbitrary implementations of `httpcache.Cache`.